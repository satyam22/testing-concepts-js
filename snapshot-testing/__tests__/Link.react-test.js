import React from 'react';
import Link from '../Link.react';
import renderer from 'react-test-renderer';

('use strict');

test('It renderes correctly', () => {
  const tree = renderer
    .create(<Link page="satyambnsal.com">Author site</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders as an anchor when no page is set', () => {
  const tree = renderer.create(<Link>Author site</Link>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('change the class when hovered', () => {
  const component = renderer.create(
    <Link page="satyambnsal.com">Author site</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
