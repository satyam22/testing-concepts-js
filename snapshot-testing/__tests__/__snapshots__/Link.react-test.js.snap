// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renderes correctly 1`] = `
<a
  className="normal"
  href="satyambnsal.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Author site
</a>
`;

exports[`change the class when hovered 1`] = `
<a
  className="normal"
  href="satyambnsal.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Author site
</a>
`;

exports[`change the class when hovered 2`] = `
<a
  className="hovered"
  href="satyambnsal.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Author site
</a>
`;

exports[`change the class when hovered 3`] = `
<a
  className="normal"
  href="satyambnsal.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Author site
</a>
`;

exports[`renders as an anchor when no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Author site
</a>
`;
